src/app/work-item/work-item-list/work-item-list.component.html

```diff
@@ -1,33 +1,33 @@
 <app-header></app-header>
 <div class="container-fluid screen-min-ht">
 	<div class="list-group list-view-pf list-view-pf-view">
-		 <div>
+		 <div id="workItemList_quickAdd">
 			 <div class="list-group-item supress-hover" (click)="addWorkItem()">
 				<work-item-quick-add (close)="close($event)"></work-item-quick-add>
 			</div>
 		</div>
-		<div *ngFor="let workItem of workItems">
-			<div class="list-group-item" (click)="onSelect(workItem)"
+		<div *ngFor="let workItem of workItems" id="workItemList_OuterWrap">
+			<div class="list-group-item" id="workItemList_Container" (click)="onSelect(workItem)"
 					 [class.selected]="workItem === selectedWorkItem">
 				<div class="list-view-pf-actions">
-					<button class="btn btn-default" (click)="gotoDetail(workItem)">View Details</button>
-					<button class="btn btn-default delete-button" (click)="deleteWorkItem(workItem); $event.stopPropagation()">Delete</button>
+					<button class="btn btn-default" id="workItemList_ViewItemDetailBtn" (click)="gotoDetail(workItem)">View Details</button>
+					<button class="btn btn-default delete-button" id="workItemList_deleteListItemBtn" (click)="deleteWorkItem(workItem); $event.stopPropagation()">Delete</button>
 					<div class="dropdown pull-right dropdown-kebab-pf">
-						<button class="btn btn-link dropdown-toggle" type="button" id="dropdownKebabRight" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
+						<button class="btn btn-link dropdown-toggle" type="button" id="workItemList_dropdownKebabRightBtn" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
 							<span class="fa fa-ellipsis-v"></span>
 						</button>
 						<ul class="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownKebabRight">
-							<li><a href="#">Delete</a></li>
-							<li><a href="#">Move to backlog</a></li>
-							<li><a href="#">Change Status</a></li>
+							<li><a href="#" id="workItemList_Delete">Delete</a></li>
```

Comment:
All elements with IDs between line numbers 10 to 50 would not be unique.
Line 9 <div *ngFor="let workItem of workItems" id="workItemList_OuterWrap"> iterates over an array so we would have the IDs repeating.
In such cases, add a dynamic counter or get the index from ngFor and append it to the IDs.